src/middlewared/middlewared/plugins/alert.py

```diff
@@ -0,0 +1,493 @@
+
+                                for name, verbose_name in (
+                                    ("name", "Contact Name"),
+                                    ("title", "Contact Title"),
+                                    ("email", "Contact E-mail"),
+                                    ("phone", "Contact Phone"),
+                                    ("secondary_name", "Secondary Contact Name"),
+                                    ("secondary_title", "Secondary Contact Title"),
+                                    ("secondary_email", "Secondary Contact E-mail"),
+                                    ("secondary_phone", "Secondary Contact Phone"),
+                                ):
+                                    value = getattr(support, name)
+                                    if value:
+                                        msg += ["", "{}: {}".format(verbose_name, value)]
+
+                                try:
+                                    await self.middleware.call("support.new_ticket", {
+                                        "title": "Automatic alert (%s)" % serial,
+                                        "body": "\n".join(msg),
+                                        "attach_debug": False,
+                                        "category": "Hardware",
+                                        "criticality": "Loss of Functionality",
+                                        "environment": "Production",
+                                        "name": "Automatic Alert",
+                                        "email": "auto-support@ixsystems.com",
+                                        "phone": "-",
+                                    })
+                                except Exception:
+                                    self.logger.error(f"Failed to create a support ticket", exc_info=True)
+
+    async def __run_alerts(self):
+        failover = False
+        if not await self.middleware.call("system.is_freenas"):
+            failover = await self.middleware.call("notifier.failover_licensed")
+
+        for alert_source in ALERT_SOURCES.values():
+            if datetime.utcnow() < self.alert_source_last_run[alert_source.name] + alert_source.interval:
+                continue
+
+            self.alert_source_last_run[alert_source.name] = datetime.utcnow()
+
+            self.logger.trace("Running alert source: %r", alert_source.name)
+
+            alerts_a = await self.__run_source(alert_source.name)
+            for alert in alerts_a:
+                alert.node = "A"
+
+            alerts_b = []
+            if failover and alert_source.run_on_passive_node:
+                try:
+                    alerts_b = [Alert(**alert)
+                                for alert in (await self.middleware.call("failover.call_remote", "alert.run_source",
+                                                                         alert_source.name))]
```

Comment:
Am I understanding it correctly it will try to run that under any condition and generate an alert if it fails?

It could be the passive node is down or is in upgrade process (running a different version).

Reply:
What is the proper check to determine if we should skip running alerts on node B?

Reply:
I dont think we have anything in place for that yet.

How about we start with "responds to core.ping, has a failover_status of BACKUP and is running the same version"?